WooCommerce/Classes/ViewRelated/Orders/Order Creation/PaymentSection/FeeOrDiscountLineDetailsViewModel.swift

```diff
@@ -111,6 +111,11 @@ final class FeeOrDiscountLineDetailsViewModel: ObservableObject {
     ///
     @Published var percentage: String = ""
 
+    /// Captures state when a discount value should be disallowed, for example,
+    /// when the discount entered is higher than the total price of a product.
+    ///
+    @Published private(set) var discountValueIsDisallowed: Bool = false
```

Comment:
I'd suggest either naming this specific to the reason for why it's disallowed, or making it an enum that explains why with code, rather than a bool.

I think at the moment it's only used for the discount being too high, so let's call it
```suggestion
    @Published private(set) var discountExceedsProductPrice: Bool = false
```

Reply:
Updated! 177128191203b9cbd92abfa18026cb158ab65cb6